frontend/src/metabase-lib/parameters/utils/parameter-source.ts

```diff
@@ -1,27 +1,49 @@
-import { ValuesSourceConfig, ValuesSourceType } from "metabase-types/api";
+import {
+  Dataset,
+  ValuesSourceConfig,
+  ValuesSourceType,
+} from "metabase-types/api";
 
 export const isValidSourceConfig = (
   sourceType: ValuesSourceType,
-  sourceConfig: ValuesSourceConfig,
+  { card_id, value_field, values }: ValuesSourceConfig,
 ) => {
   switch (sourceType) {
     case "card":
-      return sourceConfig.card_id != null && sourceConfig.value_field != null;
+      return card_id != null && value_field != null;
     case "static-list":
-      return sourceConfig.values != null && sourceConfig.values.length > 0;
+      return values != null && values.length > 0;
     default:
       return true;
   }
 };
 
-export const getDefaultSourceConfig = (
+export const getSourceConfigForType = (
   sourceType: ValuesSourceType,
-  sourceValues?: string[],
+  { card_id, value_field, values }: ValuesSourceConfig,
 ): ValuesSourceConfig => {
   switch (sourceType) {
+    case "card":
+      return { card_id, value_field };
     case "static-list":
-      return { values: sourceValues };
+      return { values };
     default:
       return {};
   }
 };
+
+const getUniqueNonNullValues = (values: unknown[]) => {
+  return Array.from(new Set(values))
+    .filter(value => value != null)
```

Comment:
should we use `isNotNull`?https://github.com/metabase/metabase/blob/d1d41d6587cd4d21b56fa12a87308f60f1b89cf1/frontend/src/metabase/core/utils/types/types.ts#L1-L3